js/coinsbit.js

```diff
@@ -0,0 +1,520 @@
+        }
+        let timestamp = undefined;
+        if ('timestamp' in order) {
+            timestamp = this.safeTimestamp (order, 'timestamp');
+        } else if ('ctime' in order) {
+            timestamp = this.safeTimestamp (order, 'ctime');
+        }
+        const datetime = this.iso8601 (timestamp);
+        const lastTradeTimestamp = this.safeTimestamp (order, 'ftime');
+        const symbol = orderMarket['symbol'];
+        const type = this.safeString (order, 'type');
+        const side = this.safeString (order, 'side');
+        const price = this.safeFloat (order, 'price');
+        const amount = this.safeFloat (order, 'amount');
+        const remaining = this.safeFloat (order, 'left');
+        let status = undefined;
+        let filled = undefined;
+        if (remaining === undefined || remaining === 0.0) {
+            status = 'closed';
+            filled = amount;
+        } else {
+            status = 'open';
+            filled = amount - remaining;
+        }
+        const cost = price * filled;
+        const fee = {
+            'currency': orderMarket['quote'],
+            'cost': this.safeFloat (order, 'dealFee'),
+        };
+        return {
+            'id': id,
+            'datetime': datetime,
+            'timestamp': timestamp,
+            'lastTradeTimestamp': lastTradeTimestamp,
+            'status': status,
+            'symbol': symbol,
+            'type': type,
+            'side': side,
+            'price': price,
+            'amount': amount,
+            'filled': filled,
+            'remaining': remaining,
+            'cost': cost,
+            'trades': undefined,
+            'fee': fee,
+            'info': order,
+        };
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        if (type !== 'limit') {
+            throw new ExchangeError (this.id + ' allows limit orders only');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+            'side': side,
+            'price': this.priceToPrecision (symbol, price),
+            'amount': this.amountToPrecision (symbol, amount),
+        };
+        const response = await this.privatePostOrderNew (this.extend (request, params));
+        const result = this.safeValue (response, 'result');
+        return this.parseOrder (result, market);
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+            'orderId': parseInt (id),
+        };
+        const response = await this.privatePostOrderCancel (this.extend (request, params));
+        const result = this.safeValue (response, 'result');
+        return this.parseOrder (result, market);
+    }
+
+    async fetchOrder (id, symbol = undefined, params = {}) {
```

Comment:
This method is redundant, the user can do the filtering on his side, we don't need to invent the methods that are not provided by the exchange, instead we mostly support what's available from the API natively and also offer filtering base methods to the user as well. So I suggest to remove all methods that call `fetchOpenOrders` in this case.

Reply:
P.S. `findOrderById` is redundant as well.